src/tests/simple_dfs.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * (C) Copyright 2020-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+/*
+ * This provides a simple example for how to get started with a DFS container.
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <daos.h>
+#include <daos_fs.h>
+
+#define FAIL(fmt, ...)						\
+do {								\
+	fprintf(stderr, fmt " aborting\n", ## __VA_ARGS__);	\
+	exit(1);						\
+} while (0)
+
+#define	ASSERT(cond, ...)					\
+do {								\
+	if (!(cond))						\
+		FAIL(__VA_ARGS__);				\
+} while (0)
+
+
+int
+main(int argc, char **argv)
+{
+	dfs_t		*dfs;
+	dfs_obj_t	*dir1, *f1;
+	int		rc;
+
+	if (argc != 3) {
+		fprintf(stderr, "usage: ./exec pool cont\n");
+		exit(1);
+	}
+
+	/** initialize the local DAOS stack */
+	rc = dfs_init();
+	ASSERT(rc == 0, "dfs_init failed with %d", rc);
```

Comment:
I realize this is a test program, but do we want to format return codes?

Reply:
sure, but don't think it's really needed in this case here. my main goal is to just point to people at a simple flow for dfs API usage.

Reply:
Makes sense. What I can see happening is:
1. User downloads this sample file
2. User tries to build
3. User wonders why our own sample is broken

But that applies to more than just THIS sample file, so not a blocker for me.

Reply:
is it broken?

Reply:
i actually tested before i commit, but i might have missed something..

Reply:
> What I can see happening is

Sorry, I meant this as a hypothetical.

No, it's not broken. And very unlikely to be since the API is supposed to be backwards compatible. But in a general sense, I have found broken example programs out in the wild before, and it can be frustrating as a new user.

Reply:
yes indeed, we should run the simple examples at least in weekly tests (simple array, object, and now dfs). i thought there was a ticket to do that assigned  to someone, but i can't find it now.

Reply:
https://daosio.atlassian.net/browse/DAOS-8279